File: year2/semester2/mpp/c#/TripClientServer/TripsServer/Net/RpcConcurrentServer.cs
using log4net;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net.Sockets;
using System.Text;
using System.Threading;
using System.Threading.Tasks;
using TripCommon.Validator;

namespace TripsServer.Net
{
    class RpcConcurrentServer : AbstractConcurrentServer
    {
        private static readonly ILog LOGGER = LogManager.GetLogger("RpcConcurrentServer");

        private ITripServices tripServices;

        public RpcConcurrentServer(string host, int port, ITripServices tripServices) : base(host, port)
        {
            this.tripServices = tripServices;
        }
        protected override Thread CreateWorker(TcpClient client)
        {
            LOGGER.InfoFormat("creating worker for client {0}", client);
            TripClientRpcWorker worker = new TripClientRpcWorker(tripServices, client);
            return new Thread(new ThreadStart(worker.Run));
        }
    }
}

File: TripClientServer/TripServer/src/main/java/net/RpcConcurrentServer.java
package net;


import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;
import service.TripServices;

import java.net.Socket;


public class RpcConcurrentServer extends AbsConcurrentServer {
    private static final Logger LOGGER = LogManager.getLogger(RpcConcurrentServer.class.getName());
    private TripServices tripServices;

    public RpcConcurrentServer(int port, TripServices tripServices) {
        super(port);
        this.tripServices = tripServices;
    }

    @Override
    protected Thread createWorker(Socket client) {
        LOGGER.info("creating TripClientRpcWorker for client " + client);
        TripClientRpcWorker worker=new TripClientRpcWorker(tripServices, client);
        Thread tw=new Thread(worker);
        LOGGER.traceExit();
        return tw;
    }
}
